psi4/src/psi4/dcft/dcft_scf_UHF.cc

```diff
@@ -290,8 +290,8 @@ void DCFTSolver::print_orbital_energies() {
         Ca_->print();
         Cb_->print();
     }
-    delete[] aIrrepCount;
-    delete[] bIrrepCount;
+    free(aIrrepCount);
+    free(bIrrepCount);
```

Comment:
strange, it was already `free` in two other files. wonder how this one was missed.

Reply:
I am pretty sure we want `delete[]` rather than `free`... I'd rather focus on getting these arrays using `new` or `std::vector` rather that `init_int_array`

Reply:
I don't follow. I think you're saying, "We want to use delete[] instead of free wherever possible because Reason X. Therefore, it would be better to change aIrrepCount and bIrrepCount to being created through new or std::vector rather than through init_int_array, so that delete[] is the correct call." Do I have that right?

Reply:
The arrays are allocated using `malloc` in `init_int_array`. LGTM is right in complaining because the corresponding deallocation should be done using `free`. However, these arrays can be just as easily created either with:
- `std::vector`, which would free resources automatically, or 
- `auto aIrrepCount = new  int[nIrrep_];`, which is the idiomatic C++ construct. The deallocation would then be done with `delete[] aIrrepCount;`.

For primitive types there is no difference, that I know of, between using the `malloc`/`free` couple and the `new[]/delete[]` couple. Just a question of being idiomatic in C++.

`init_int_array` also zeros out the newly allocated vector. This achieves the same: `auto aIrrepCount = std::vector<int>(nIrrep_, 0);`

Reply:
Okay, so why not replace _all_ the `init_int_array` calls in `dcft`? By my count, there are 22 total. (And it looks like some of them are never freed. This includes `aocc_off_` at least and probably others.)

Reply:
I have nothing against the change you propose. Doesn't LGTM complain about the arrays that don't get freed?

Reply:
Apparently not.

The `init_int_array` calls fell into three classes.
1. `aIrrepCount` and friends. Changed to `std::vector`.
2. `aocc_off_` and friends. These are needed by libdpd, which cannot take a `std::vector`, so they are now called by `new int[nirrep_]` syntax. They are also properly deleted.
3. The `dcft_sort_mo_tpdm.cc` functions. This is a complicated enough function that I didn't want to touch it. Especially as I have plans to eliminate this in the near future. I cannot imagine why RHF-DCT needs its own TPDM sort for gradients.

This should satisfy the changes?

Reply:
Thanks for your thorough work!